### PR TITLE
qusb2snes: init at 0.7.19.1

### PR DESCRIPTION
**Pull request by IslandUsurper**
###### Motivation for this change
Enable auto-tracking, netplay, etc for SNES emulators.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by neonfuz** [comment]
I took a quick look at the changes, it all looks good but I suggest using fetchFromGitHub over fetchTarball when fetching from github, idr why but it's preferred (something like faster downloads).

File: pkgs/applications/networking/qusb2snes/default.nix [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{ lib, mkDerivation, git, qmake, qtbase, qtserialport, qtwebsockets }:
+mkDerivation rec {
+  pname = "QUsb2snes";
+  version = "0.7.19.1";
+
+  src = fetchTarball {
+    url = "https://github.com/Skarsnik/QUsb2snes/archive/v${version}.tar.gz";
+    sha256 = "1fj09sj072j3055dfjfbhzil0hqz2bv3mwv09477iz17bvifhf3k";
+  };
```
Comment (IvarWithoutBones) [inline pkgs/applications/networking/qusb2snes/default.nix:9]: You should be using `fetchFromGitHub`.
```suggestion
  src = fetchFromGitHub {
    owner = "Skarsnik";
    repo = "QUsb2snes";
    rev = "v${version}";
    sha256 = "0000000000000000000000000000000000000000000000000000000000000000";
  };
```

File: pkgs/applications/networking/qusb2snes/default.nix [hunk 2]
```diff
@@ -0,0 +1,20 @@
+{ lib, mkDerivation, git, qmake, qtbase, qtserialport, qtwebsockets }:
+mkDerivation rec {
```
Comment (IvarWithoutBones) [inline pkgs/applications/networking/qusb2snes/default.nix:2]: Minor spacing thing.
```suggestion
{ lib, mkDerivation, git, qmake, qtbase, qtserialport, qtwebsockets }:

mkDerivation rec {
```

File: pkgs/applications/networking/qusb2snes/default.nix [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+  buildInputs = [ git qtbase qtserialport qtwebsockets ];
+  nativeBuildInputs = [ qmake ];
+
+  qmakeFlags = [ "${src}/QUsb2snes.pro" ];
+
+  installPhase = ''
+    mkdir -p $out/bin
+    cp ./QUsb2Snes $out/bin
+  '';
```
Comment (IvarWithoutBones) [inline pkgs/applications/networking/qusb2snes/default.nix:19]: A `meta` section is missing.

**Review by IvarWithoutBones — CHANGES_REQUESTED** [review]
A few minor problems, otherwise LGTM :)

File: pkgs/applications/networking/qusb2snes/default.nix (hunk 1 above)
Comment (IslandUsurper) [reply pkgs/applications/networking/qusb2snes/default.nix]: Ha, didn't expect the hash to be the same, but these two calls seem equivalent in result.

**Comment by IslandUsurper** [comment]
Thanks for the reviews. All suggestions have been implemented. Sorry for forgetting `meta`.

File: pkgs/applications/networking/qusb2snes/default.nix [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+  meta = with lib; {
+    description = "A Qt based webserver for usb2snes";
+    license = licenses.gpl3Only;
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/qusb2snes/default.nix:26]: homepage missing.

**Comment by SuperSandro2000** [comment]
Closing due to inactivity from author.

**Pull request closed without merging**